syntax = "proto3";

//The .proto file starts with a package declaration, which helps to prevent naming conflicts between different projects.
package chaosmonkey;

service ChaosMonkey {
    rpc UploadMatrix(ConnMatrix) returns (Status) {}
    rpc UpdateValue(MatValue) returns (Status) {}
}

enum StatusCode {
    OK = 0;
    ERROR = 1;
}

message Status {
    StatusCode ret = 1;
}

message ConnMatrix {
    message MatRow {
        repeated float vals = 1;
    }
    repeated MatRow rows = 1;
}

message MatValue {
    int32 row = 1;
    int32 col = 2;
    float val = 3;
}